src/app/CommandSender.cpp

```diff
@@ -68,6 +68,16 @@ CommandSender::CommandSender(Callback * apCallback, Messaging::ExchangeManager *
     assertChipStackLockedByCurrentThread();
 }
 
+CommandSender::CommandSender(ExtendedCallback * apExtendedCallback, Messaging::ExchangeManager * apExchangeMgr,
+                             bool aIsTimedRequest, bool aSuppressResponse) :
+    mExchangeCtx(*this),
+    mpExtendedCallback(apExtendedCallback), mpExchangeMgr(apExchangeMgr), mSuppressResponse(aSuppressResponse),
+    mTimedRequest(aIsTimedRequest)
+{
+    assertChipStackLockedByCurrentThread();
```

Comment:
This is this needed in the first place?

Reply:
Other constructors have it so I don't want to make that change here. If we want this to be created without ChipStackLock then the removal can be done in a separate PR that focuses just on that. This PR is already large enough